app/lib/katello/validators/kickstart_repository_validator.rb

```diff
@@ -0,0 +1,33 @@
+module Katello
+  module Validators
+    class KickstartRepositoryValidator < ActiveModel::Validator
+      def validate(record)
+        ks_repo_id = kickstart_repository_id(record)
+        return unless ks_repo_id
+
+        msg = if !record.operatingsystem.blank?
+                _("Please select an operating system before assigning a kickstart repository")
+              elsif !record.architecture.blank?
+                _("Please select an architecture before assigning a kickstart repository")
+              elsif !record.operatingsystem.is_a?(Redhat)
+                _("Kickstart repositories can only be associated with Red Hat hosts")
+              elsif record.content_source.blank?
+                _("Please select a content source before assigning a kickstart repository")
+              elsif record.operatingsystem.kickstart_repos(record).none? { |repo| repo[:id] == ks_repo_id }
+                _("No such repository in the assigned content view, environment, content source, OS, and architecture")
```

Comment:
Might need to work on the messaging here. May be something along "the kickstart repository/distribution assigned to this entity is not part of the entity's content view, lifecycle environment, content source, operating and architecture"
This is because the user who does something like
```
hammer hostgroup update --content-view-id=100 --id=1
```

will end up seeing the above error and it might not be entirely obvious why there is an error about the kickstart repository when one is updating the content-view-id.